lib/MetagenomeAPI/MetagenomeAPIImpl.py

```diff
@@ -252,6 +272,85 @@ def search_region(self, ctx, params):
                              'result is not type dict as required.')
         # return the results
         return [result]
+
+    def search_contigs(self, ctx, params):
+        """
+        :param params: instance of type "SearchContigsOptions" -> structure:
+           parameter "ref" of String, parameter "start" of Long, parameter
+           "limit" of Long, parameter "sort_by" of list of type
+           "column_sorting" -> tuple of size 2: parameter "column" of String,
+           parameter "ascending" of type "boolean" (Indicates true or false
+           values, false = 0, true = 1 @range [0,1])
+        :returns: instance of type "SearchContigsResult" -> structure:
+           parameter "num_found" of Long, parameter "start" of Long,
+           parameter "contigs" of list of type "contig" -> structure:
+           parameter "contig_id" of String, parameter "feature_count" of
+           Long, parameter "length" of Long
+        """
+        # ctx is the context object
+        # return variables are: result
+        #BEGIN search_contigs
+        if 'ref' not in params:
+          raise RuntimeError(f"'ref' argument required for search_contigs")
+        if 'limit' not in params:
+          raise RuntimeError(f"'limit' argument required for search_contigs")
+        if 'start' not in params:
+          raise RuntimeError(f"'start' argument required for search_contigs")
+        if not params.get('sort_by'):
+          sort_by = ('length', 0)
+        else:
+          sort_by = params['sort_by']
+          if sort_by[0] not in ['length', 'contig_id', 'feature_count']:
+            raise RuntimeError("'sort_by' argument can only contain one "
+                               "of 'length', 'contig_id', or 'feature_count'")
+
+        ws = Workspace(self.config['workspace-url'], token=ctx['token'])
+        ama_utils = AMAUtils(ws)
+        params['included_fields'] = ['contig_ids', 'contig_lengths']
+        ama = ama_utils.get_annotated_metagenome_assembly(params)['genomes'][0]
+        data = ama['data']
+        if len(data['contig_ids']) != len(data['contig_lengths']):
+          raise RuntimeError(f"contig ids (size: {len(data['contig_ids'])}) and contig "
+                             f"lengths (size: {len(data['contig_lengths'])}) sizes do not match.")
+        contig_ids = data['contig_ids']
+        contig_lengths = data['contig_lengths']
+
+        if sort_by[0] == 'contig_id' and not sort_by[1]:
```

Comment:
`not sort_by[1]` will only work if your param is ["x", None], but it will throw an indexing error on just ["x"]. You might instead want to do `if sort_by[0] == 'contig_id' and len(sort_by) == 1`